### ISSUE TITLE
emacs not receiving Ctrl-Space on windows

### ISSUE THREAD

**Issue by mammothbane** [zeit/hyper#2016]
- [x] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version (**I'm on master**)
- [x] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate

- **OS version and name**: Windows 10 Home v1607
- **Hyper.app version**: master @ 8a23b18 (v1.3.3)
- **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**: https://gist.github.com/mammothbane/1001baea760ca84fef09e4573d67af0b
- **Relevant information from devtools**: N/A
- **The issue is reproducible in vanilla Hyper.app**: yes

## Issue
With #1509's children merged, I would expect to see Ctrl-space captured by hyper, but emacs doesn't set the mark when I type `Ctrl-space`. I've verified that emacs has the binding. 

I set another binding that I *can* use, but this is the default binding for `set-mark-command`, and it doesn't work for me.

**Comment by ppot** [comment]
What does `Ctrl-space` does? Can you make a gif?

**Comment by mammothbane** [comment]
Currently, it doesn't do anything (visually, at least). No bell either.

Here it is not working on Hyper: [![asciicast](https://asciinema.org/a/0q32NBkO7lmZ4ERy0SZoWDoTe.png)](https://asciinema.org/a/0q32NBkO7lmZ4ERy0SZoWDoTe?t=6)

And here's PuTTY working fine: [![asciicast](https://asciinema.org/a/mDJCfWncVYpsVoBqNOSkU9W0Y.png)](https://asciinema.org/a/mDJCfWncVYpsVoBqNOSkU9W0Y?t=2)

**Comment by incomingstick** [comment]
I can confirm I am having the same issue. The Emacs CTRL+SPC command works on PuTTY, Cygwin, and even the default CMD terminal. I suspect it is a keybindings issue with the terminal itself, as a number of my CTRL and ALT commands do not seem to work properly either, whereas they work elsewhere.

**Comment by 4nthonylin** [comment]
Same issue, would like to bind ctrl+space to a command but it seems like the terminal it catching the key sequence. Using showkey -a yields nothing when typing ctrl+space

**Comment by chughtapan** [comment]
Any updates on this ?

**Comment by srl295** [comment]
Confirmed on Mac also with Hyper 2.0.0

**Comment by c0ze** [comment]
Confirmed on Windows Linux Subsystem and waiting for a fix. (Not only CTRL+Space, some other bindings seem to be cought too. I also couldn't CTRL+\ or CTRL+_ (undo))

**Comment by tstackhouse** [comment]
Also seeing this in Hyper 2.1.0-canary.2  Glad I'm not the only one.

**Comment by sukima** [comment]
Oddly, <kbd>CTRL</kbd>+<kbd>Space</kbd> works for me with VIM, TMUX, and/or SSH (any combination) but <kbd>CTRL</kbd>+<kbd>6</kbd> (`CTRL^`) does nothing.

Why is Hyper gobbling up <kbd>CTRL</kbd> sequences?! Sigh, have to go back to terminator, again. :cry:

**Comment by vdegenne** [comment]
I have the same issue (`hyper 3.0.0`)

Also I would like to mention something nobody mentioned, <kbd>CTRL</kbd>+<kbd>Space</kbd> works fine in `Hyper` if I open Emacs locally but doesn't work if I connect to my server using `ssh` and use Emacs.
In other words the issue is ssh-related. I believe the keystroke is not sent over a socket.

Any updates ?

**Comment by computersarecool** [comment]
Yeah Ctrl-Space works in Emacs on the Windows Ubuntu application.  But does not work in Hyper. Why is this? Is it a challenge to just pass the key input sequences?

Is this issue being worked on?

**Comment by incomingstick** [comment]
@computersarecool When you run Ubuntu on Windows via WSL it is still running in the same "terminal" application as cmd. Anything that works with that the Windows integrated terminal, should work when running the Ubuntu "app" (its just Bash being run without the extra cmd shell layer).

The reason this differs from Hyper, is because Hyper is an Electron application, and thus never touches the Windows integrated terminal. Its two completely separate systems. (That's _technically_ not true, but then you are literally debating Windows application internals).

Microsoft just did a great talk on this with their announcement of a new terminal. Its a great talk and I recommend everyone check it out: https://www.youtube.com/watch?v=KMudkRcwjCw

**Comment by computersarecool** [comment]
@Stanzilla I would either remove the platform: windows label or add the platform:mac label as well (I bet this issue exists on Linux too).

@ppot you handled a lot of the keybinding issues - do you know how we could work around this?

Similarly- how do we tell if an issue is actually being worked on - is there a way to track the hyper development?

**Comment by Stanzilla** [comment]
@computersarecool removed the win label. I don't think anyone is currently working on Hyper, so feel free to take a stab at it!

**Comment by computersarecool** [comment]
@Stanzilla do you think this project abandoned?

**Comment by computersarecool** [comment]
@Stanzilla hopefully not totally abandoned...

**Comment by mguthaus** [comment]
Seems to be abandoned. I'm also having this problem.

**Comment by mraxilus** [comment]
Adding +1 to the list of users who've encountered the issue.
